scripts/check-version.ts

```diff
@@ -0,0 +1,93 @@
+import execa = require('execa');
+import versionCommand = require('@lerna/version');
+import * as path from 'path';
+
+const errors: string[] = [];
+const main = async () => {
+  // Todo: add support for preId and distTag
+  const preId = process.argv[2];
+  const distTag = process.argv[3];
+
+  const versionChanges = await bumpVersions(preId, distTag);
+  for (const packageName of Array.from(versionChanges.keys())) {
+    const pkgVersion = versionChanges.get(packageName);
+    const gitTagName = await getGitTagName(packageName, pkgVersion);
+    const versionPublished = await doesVersionExist(packageName, pkgVersion);
+    if (versionPublished) {
+      errors.push(`${packageName}@${pkgVersion} is already published to npm`);
+      continue;
+    }
+    if (await checkIfTagExists(gitTagName)) {
+      errors.push(`${packageName}@${pkgVersion} tag already exists`);
+    }
+  }
+  if (errors.length) {
+    console.log(errors.join('\n'));
+    process.exit(1);
+  }
+};
+
+const bumpVersions = async (preid?: string, distTag?: string): Promise<Map<string, string>> => {
+  let versionChanges;
+
+  await new Promise((resolve, reject) => {
+    const config: any = {
+      conventionalCommits: true,
+      exact: true,
+      composed: 'publish',
+      lernaVersion: '4.0.0',
```

Comment:
we might want to read this version from package.json, otherwise we should document that this needs to be updated in order to update lerna

Reply:
That parameter is not needed. Removed it